src/components/filters/DateTimeFilter.vue

```diff
@@ -99,32 +97,61 @@ export default Vue.extend({
       const date = this.dateRange
       if (!date.startDate || !date.endDate) return 'Select...'
       if (date.startDate.toISOString() === date.endDate.toISOString()) {
-        return date.startDate.toLocaleString()
+        return this.beautifyDateTime(date.startDate)
       } else {
-        return `${date.startDate.toLocaleString()} - ${date.endDate.toLocaleString()}`
+        return `${this.beautifyDateTime(date.startDate)} - ${this.beautifyDateTime(date.endDate)}`
       }
+    },
+    pickerFormat () {
+      if (this.time) return 'yyyy-mm-dd HH:MM:ss'
+      else return 'yyyy-mm-dd'
     }
   },
   watch: {
     value (newValue) {
-      this.dateRange = newValue
+      this.dateRange.startDate = this.createDateFromValue(this.value[0])
+      this.dateRange.endDate = this.createDateFromValue(this.value[1])
     }
   },
+  beforeMount () {
+    this.dateRange.startDate = this.createDateFromValue(this.value[0])
+    this.dateRange.endDate = this.createDateFromValue(this.value[1])
+  },
   methods: {
     clearValue: function () {
       this.dateRange = emptyDateRange
       this.$emit('input', undefined)
     },
     updateValues: function () {
-      this.$emit('input', this.dateRange)
+      this.$emit('input', [this.dateRange.startDate, this.dateRange.endDate])
+    },
+    createDateFromValue (value) {
+      if (value) {
+        if (!isNaN(value)) return new Date(value)
+        return new Date(Date.parse(value))
+      }
+    },
+    beautifyDateTime (dateTime) {
+      if (this.time) return this.dateTimeString(dateTime)
+      else return this.dateString(dateTime)
+    },
+    dateString (dateTime) {
+      return dateTime.toLocaleDateString()
+    },
+    dateTimeString (dateTime) {
+      return dateTime.toLocaleString()
```

Comment:
Hier heb je 2 one-line functies die alleen in `beautifyDateTime` gebruikt lijken te worden. Het is leesbaarder als je die functie-calls gelijk doet in regel 135 en 136

Reply:
that makes more sense, indeed